### ISSUE TITLE
[Android] New Plugin Version doesn't work with 4.2 (API 17) and below

### ISSUE THREAD

**Issue by RonnySchleicher** [becvert/cordova-plugin-device-name#7]
### Systems:
* Windows 10 Pro 64 bit, Core i7, 8GB RAM, all updates installed with Android Studio 2.3.3, Cordova 6.5
* iMac (Retina 5K, 27 Zoll), Core i7, 16GB RAM, macOS Sierra 10.12.6, all updates installed with Android Studio 2.3.3, Cordova 6.5
### Steps to Reproduce
* create a fresh new Android Project, only with the cordova-plugin-device-name
* use follow script (in this sample for Windows)
```
call cordova create PhoneGap_android de.test.test Test
cd PhoneGap_android
call cordova platform add android
call cordova plugin add cordova-plugin-device-name
call cordova plugin ls
```

Open the Project in the Android Studio and add follow line in the File

C:\Test\PhoneGap_android\platforms\android\assets\www\js\index.js

``` Java
    // Update DOM on a Received Event
    receivedEvent: function(id) {
        var parentElement = document.getElementById(id);
        var listeningElement = parentElement.querySelector('.listening');
        var receivedElement = parentElement.querySelector('.received');

        listeningElement.setAttribute('style', 'display:none;');
        receivedElement.setAttribute('style', 'display:block;');

        console.log('Received Event: ' + id);

        //added from me to test the deviceName plugin
        console.log(cordova.plugins.deviceName.name);
    }
```

Now start the app on a real Device, in this sample i use the Emualtor to show the problem easier

Use Android 5.0 or higher, also is fine:

### API Level 21

![api21](https://user-images.githubusercontent.com/1914203/31175212-ca4e30a2-a90e-11e7-837e-ddbbc461631b.jpg)

**But use Android lower then 5, follow error occure and it is impossible to detect the devicename:**

### API Level 16, 17, 18

![image](https://user-images.githubusercontent.com/1914203/31173702-db2b3074-a908-11e7-9cd3-511448b0ca63.png)

![image](https://user-images.githubusercontent.com/1914203/31173708-e1c1f2c4-a908-11e7-8fee-adc18c781bcf.png)

![image](https://user-images.githubusercontent.com/1914203/31173716-e7a0c2f6-a908-11e7-8e86-0ab6554fcdb7.png)

**Comment by becvert** [comment]
Android emulator does not have bluetooth capabilities.
But it's raising a good point about the relevance of the alert dialog!

**Comment by RonnySchleicher** [comment]
The problem is also real devices. If you use a real Android 4.1 or 4.2 devices, the error occurs the same way. It is no longer possible to determine the name with the new version of the plugin on these devices.

**Comment by becvert** [comment]
OK...

From API 25 onward it returns [device_name](https://developer.android.com/reference/android/provider/Settings.Global.html#DEVICE_NAME)
Below 25 it uses bluetooth_name from [BluetoothManagerService.java](https://android.googlesource.com/platform/frameworks/base/+/920ace0bbc2d4133dbec991d2636c99a57d6245e/services/java/com/android/server/BluetoothManagerService.java#40)
Probably that doesn't work below 18 (4.3) because [BluetoothManager](https://developer.android.com/reference/android/bluetooth/BluetoothManager.html) did not exist.

Previous versions of this plugin used [BluetoothAdapter.getName](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#getName()) wich worked since API 5 but required BLUETOOTH permission.

My motivation was to remove that permission.

For my applications for instance, I build a package for API 16 to 19 and another package for API 21 and above. For the latter I do not want to add the permission.

Maybe I'll revert back to using BluetoothAdapter by default and provide a switch allowing the new method instead.

**Comment by RonnySchleicher** [comment]
It would solve in a plugin with a switch would be the perfect solution.  :thumbsup:

**Comment by becvert** [comment]
Please upgrade to 1.3.1

Reverted to using BLUETOOTH permission for devices with API 17 and below.
I found out the `android:maxSdkVersion` attribute of `uses-permission` and set it to 17, so above the system will no longer grant the permission.

I also removed the use of `device_name` for now because I'm not guaranteed it is a user friendly name after all. it might just be the model of the device. It needs further investigation before I put it back in.
That means a bluetooth adapter is required again for API 25+.

I'm still unsure about what to do when the device name is not found because of no adapter.
An alert pops up at the moment. shall I just allow an empty value or return a default value like the device's model...?

**Comment by RonnySchleicher** [comment]
Great! This all sounds very good!

I personally would return an empty string if no device name can be obtained.
The current message is a very bad solution. If an empty string is returned, I can decide in my application what to do. If a message box is displayed, this is not possible.

**Comment by RonnySchleicher** [comment]
In the current Plugin Version is a configuration Error. I cannot upload and signed apk the the Play Store.
Follow message appears:

![image](https://user-images.githubusercontent.com/1914203/31558157-0b822990-b04c-11e7-8ff4-52a11303843f.png)

**Comment by RonnySchleicher** [comment]
The Message in English:
**The maxSdkVersion value for the android.permission.BLUETOOTH authority must be at least 18.**

**Comment by RonnySchleicher** [comment]
I hav fixed it for me, but it also should be fixed in your repository.

**Comment by becvert** [comment]
fixed in 1.3.2

**Comment by RonnySchleicher** [comment]
Thank you very much! Works perfect!
